Allow multiple connections to be passed to haproxy across the the reverseproxy relation

digitalrane opened the pull request:
This allows the haproxy charm to take advantage of the multicfg branch version of interface-reverseproxy, in preparation for that becoming the default behaviour. The changes in summary allow a charm to request multiple backends for a single charm. An example use case is GitLab, where both HTTP and SSH are forwarded from the load balancer to the backend. Simply forwarding HTTP would leave SSH unavailable and unroutable from the public internet, assuming the common use-case of provisioning external public access only to the load balancer.

chris-sanders committed:
Try the multicfg branch of interface-reverseproxy

chris-sanders committed:
Change config processing to accept a list of configs

chris-sanders committed:
Remove python3 string format, and add a two config unit test

Commit:
Fixed multi-port relations for TCP backends

Commit:
Added default L4 checks to backend ports

Commit:
Add mode to backend statements

chris-sanders committed:
Fix available_for_http tcp dectection

 tcp mode split to keyword/value in this branch, updated
 available_for_http to recognize this. Unit test pre-existing and passes
 with the update.

 Misc pep indent fixes for over-indention

chris-sanders commented:
I know this was first but if/when it's ready can we rebase it on #6 so I can functional test?

Commit:
Small linting fix for regex

digitalrane commented:
Rebased, reworked and passing tests. This MP contains the fixes to multi-config relations, as well as adding a default healthcheck duration to backends, where previously there was none. Includes some linting fixes as well.

chris-sanders committed:
update layer-version to latest master with config option support

Commit:
Moved to master branch of interface-reverseproxy now that multiple-port relations have landed

chris-sanders commented:
+1 I'll push this as the new stable version in the charmstore after merging.

chris-sanders reviewed (commented):
+1
